shim/src/evaluators.js

```diff
@@ -0,0 +1,35 @@
+import { evaluate } from './evaluate';
+import { setPrototypeOf } from './commons';
+
+function getEvalEvaluator(sandbox) {
+  const o = {
+    // trick to set the name of the function to "eval"
+    eval: function(sourceText) {
+      // console.log(`Shim-Evaluation: "${sourceText}"`);
+      return evaluate(sourceText, sandbox);
+    }
+  };
+  setPrototypeOf(o.eval, sandbox.Function);
+  o.eval.toString = () => 'function eval() { [shim code] }';
+  return o.eval;
+}
+
+function getFunctionEvaluator(sandbox) {
+  const { confinedWindow } = sandbox;
+  const f = function Function(...args) {
+    // console.log(`Shim-Evaluation: Function("${args.join('", "')}")`);
+    const sourceText = args.pop();
+    const fnArgs = args.join(', ');
```

Comment:
This is what we did in SES but I'm no longer convinced it is safe. More robust would be to also verify that each parameter is an identifier, or that the fnArgs is a comma separated list of identifiers and whitespace. What does the spec say about modern parameter syntax (patterns, defaults, rest) here?